projects/distributed-tracing/src/shared/dashboard/widgets/table/table-widget-renderer.component.ts

```diff
@@ -69,33 +97,36 @@ export class TableWidgetRendererComponent
 
     this.metadata$ = this.getScopeAttributes();
     this.columnConfigs = this.getColumnConfigs();
+
+    this.initControls();
+  }
+
+  private initControls(): void {
+    this.filterItems = this.api.model.filterToggles.map(filter => ({
+      label: capitalize(filter.label),
+      value: filter
+    }));
+    this.modeItems = this.api.model.modeToggles.map(mode => ({
+      label: capitalize(mode),
+      value: mode
+    }));
   }
 
   public getChildModel = (row: TableRow): object | undefined => this.model.getChildModel(row);
 
-  public onRowSelection(selections: StatefulTableRow[]): void {
-    if (this.api.model.selectionMode === TableSelectionMode.Single) {
-      /**
-       * Execute selection handler for single selection mode only
-       */
-      let selectedRow;
-      if (selections.length > 0) {
-        selectedRow = selections[0];
-        this.selectedRowInteractionHandler = this.getInteractionHandler(selectedRow);
-      }
+  protected onDashboardRefresh(): void {
+    this.initControls();
```

Comment:
So this is where it's used - why does initControls need to be invoked on a dashboard refresh? The filter items and mode items are the same. Is the idea to reset their value? If so, I'd actually assume the opposite behavior, that they'd persist their state through a local refresh.

Reply:
Fixed